tests/Admin/Admin_Page_Tests.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Tests for the Admin_Page class.
+ *
+ * @package plugin-check
+ */
+
+namespace Admin;
+
+use WordPress\Plugin_Check\Admin\Admin_Page;
+use WP_UnitTestCase;
+
+class Admin_Page_Tests extends WP_UnitTestCase {
+
+	protected $admin_page;
+
+	public function set_up() {
+		parent::set_up();
+		$this->admin_page = new Admin_Page();
+	}
+
+	public function test_add_hooks() {
+		$this->admin_page->add_hooks();
+		$this->assertEquals( 10, has_action( 'admin_menu', array( $this->admin_page, 'add_page' ) ) );
+	}
+
+	public function test_add_page() {
+		global $_parent_pages;
+
+		$current_screen = get_current_screen();
+
+		$admin_user = self::factory()->user->create( array( 'role' => 'administrator' ) );
+
+		if ( is_multisite() ) {
+			grant_super_admin( $admin_user );
+		}
+
+		wp_set_current_user( $admin_user );
+		set_current_screen( 'dashboard' );
+
+		$this->admin_page->add_page();
+
+		$this->assertArrayHasKey( 'plugin-check', $_parent_pages );
+		$this->assertEquals( 'tools.php', $_parent_pages['plugin-check'] );
+
+		set_current_screen( $current_screen );
```

Comment:
This does not work reliably - if an assertion above fails, this code doesn't run.

We need to assign the `$_parent_pages` global to a local variable _before_ doing the assertions, then reset `set_current_screen`, and only at the very end run the assertions on the local variable.

Reply:
Done